src/target/riscv_debug.c

```diff
@@ -834,6 +835,7 @@ static void riscv_halt_resume(target_s *target, const bool step)
 		stepping_config |= RV_DCSR_STEP | RV_DCSR_STEPIE;
 	else
 		stepping_config &= ~(RV_DCSR_STEP | RV_DCSR_STEPIE);
+	stepping_config |= RV_DCSR_EBREAKM;
```

Comment:
This looks almost correct - not sure it's semantically correct to be enabled when we're doing single stepping though so it might need pushing into the else statement here as in:
```c
	else {
		stepping_config &= ~(RV_DCSR_STEP | RV_DCSR_STEPIE);
		stepping_config |= RV_DCSR_EBREAKM;
	}
```

We'll go read the debug spec to double check this though and re-review accordingly.

Reply:
Ok, having read the specs - this looks good, though we stand by our previous observation around single stepping. Perhaps the only small change we'd make is to rename the macro from `EBREAKM` to `EBREAK_MACHINE` to indicate what the `m` means - or at least `EBREAK_M` to indicate it's intentional. The reason for this is we should probably also definedn `RV_DCSR_EBREAK_SUPERVISOR` and `RV_DCSR_EBREAK_USER` macros for the other two modes this can be done for.

Reply:
Thank you
Normally i've taken the remarks into account & squash the commits.

next are flashstub for riscv & wch chip supports.